lib/http.js

```diff
@@ -31,12 +31,46 @@ exports.request = function(options, cb) {
   return new ClientRequest(options, cb);
 };
 
+exports.requestAsync = function(options) {
+  var resolveRequest = null;
+  const getResponse = new Promise((resolve) => {
+    resolveRequest = resolve;
+  });
+  const request = exports.request(options, (res) => {
+    resolveRequest(res);
+  });
+  return {
+    request,
+    response: getResponse
+  };
+};
+
 exports.get = function(options, cb) {
   var req = exports.request(options, cb);
   req.end();
   return req;
 };
 
+exports.getAsync = function(options) {
+  var resolveRequest = null;
+  var rejectRequest = null;
+  const getResponse = new Promise((resolve, reject) => {
+    resolveRequest = resolve;
+    rejectRequest = reject;
+  });
+  const request = exports.request(options, (res) => {
+    resolveRequest(res);
+  });
+  request.on('error', rejectRequest);
+  getResponse.then(() => {
```

Comment:
It doesn't play well with 'maxListeners' limit. It will be confusing if you set 'maxListeners' to 10 and get warning on 9th listener in user code. It's an edge case, but still...